http-service/src/main/java/net/runelite/http/service/chat/ChatController.java

```diff
@@ -250,9 +252,12 @@ public void submitLayout(@RequestParam String name, @RequestBody LayoutRoom[] ro
 	}
 
 	@PostMapping("/pets")
-	public void submitPetList(@RequestParam String name, @RequestBody int[] petList)
+	public void submitPetList(@RequestParam String name, @RequestBody HashMap<Integer, Integer> petList)
 	{
-		if (petList.length == 0 || petList.length > MAX_PETS)
+		if (petList.size() == 0
+			|| petList.size() > MAX_PETS
+			|| Collections.max(petList.values()) > MAX_PETS
+			|| petList.values().stream().reduce(0, Integer::sum) > MAX_PETS)
```

Comment:
Here also, why would it matter if a user has received enough duplicate pets that across all pets they had gotten more than 256 pet rolls?

Reply:
For data validation. The check may be redundant as filling the chatbox with 250-ish icons might be stretching it, so I can remove this